### ISSUE TITLE
 Cowboy handling of X-Forwarded-Proto

### ISSUE THREAD

**Issue by aShobhitSingh** [ninenines/cowboy#1025]
How does Cowboy handles X-Forwarded-Proto. I searched into the code and it seems that it is not being handled. What I wanted to do is terminate the ssl at my load balancer and then Cowboy will receive this header from LB.
How I can tell Cowboy of initial https connection. Any help would be appreciated

**Comment by essen** [comment]
It doesn't, and I've not heard of it before. Do you have a link to its specification?

**Comment by bltb** [comment]
https://tools.ietf.org/html/draft-ietf-appsawg-http-forwarded-10#section-5.4

**Comment by essen** [comment]
Ah, same as Forwarded. Thanks. In that case this link is more up to date: https://tools.ietf.org/html/rfc7239#section-5.4

Current plans for this RFC are to implement the parsing of the header(s) and nothing else. So for x-forwarded-proto it probably wouldn't do much other than validating the syntax to be a valid scheme.

The reason for not wanting to add any Forwarded logic is explained in that same RFC (https://tools.ietf.org/html/rfc7239#section-8.1). I would probably be OK with a helper middleware to do the white listing and update the Req accordingly but none of this should be enabled by default.

**Comment by aShobhitSingh** [comment]
Hope I understood correctly, what you are saying. It means, it will not be handled internally, but from the request handler part it can check this header's value to look for the initial secure connection.

**Comment by essen** [comment]
Yes the header is there, Cowboy will just do nothing about it by default. Cowboy will only aim to provide functions to parse the header(s).
